src/renderer/page/subscriptions/view.jsx

```diff
@@ -1,83 +1,149 @@
   }
 
-  render() {
-    const { subscriptions, subscriptionClaims, loading, autoDownload } = this.props;
+  renderSubscriptions() {
+    const { viewMode, unreadSubscriptions, allSubscriptions } = this.props;
 
-    let claimList = [];
-    subscriptionClaims.forEach(claimData => {
-      claimList = claimList.concat(claimData.claims);
-    });
+    if (viewMode === VIEW_ALL) {
+      return (
+        <React.Fragment>
+          <div className="card__title">{__('Your subscriptions')}</div>
+          <FileList hideFilter sortByHeight fileInfos={allSubscriptions} />
+        </React.Fragment>
+      );
+    }
+    return (
+      <React.Fragment>
+        {unreadSubscriptions.length ? (
+          unreadSubscriptions.map(({ channel, uris }) => {
+            const channelLabel = channel.slice(`lbry://`.length, channel.indexOf('#'));
+            return (
+              <section key={channel}>
+                <div className="card__title">
+                  <Button
+                    button="link"
+                    navigate="/show"
+                    navigateParams={{ uri: channel }}
+                    label={channelLabel}
+                  />
+                </div>
+                <div className="card__list card__content">
+                  {uris.map(uri => <FileCard isNew key={uri} uri={uri} />)}
+                </div>
+              </section>
+            );
+          })
+        ) : (
+          <div className="page__empty">
+            <h3 className="card__title">{__('You are all caught up!')}</h3>
+            <div className="card__actions">
+              <Button button="primary" navigate="/discover" label={__('Explore new content')} />
+            </div>
+          </div>
+        )}
+      </React.Fragment>
+    );
+  }
 
-    const subscriptionUris = claimList.map(claim => `lbry://${claim.name}#${claim.claim_id}`);
+  render() {
+    const {
+      subscribedChannels,
+      allSubscriptions,
+      loading,
+      autoDownload,
+      viewMode,
+      doSetViewMode,
+    } = this.props;
 
     return (
-      <Page notContained loading={loading}>
-        <HiddenNsfwClaims uris={subscriptionUris} />
-        <FormRow alignRight>
-          <FormField
-            type="checkbox"
-            name="auto_download"
-            onChange={this.onAutoDownloadChange}
-            checked={autoDownload}
-            prefix={__('Automatically download new content from your subscriptions')}
-          />
-        </FormRow>
-        {!subscriptions.length && (
+      // Only pass in the loading prop if there are no subscriptions
+      // If there are any, let the page update in the background
+      // The loading prop removes children and shows a loading spinner
+      <Page notContained loading={loading && !subscribedChannels}>
+        <HiddenNsfwClaims
+          uris={allSubscriptions.map(
+            ({ name, claim_id: claimId }) => name && claimId && `lbry://${name}#${claimId}`
```

Comment:
This is not very intuitive, I would suggest using a reduce; the logic reads very odd since it looks to be a boolean operation but returns a value.

Reply:
Maybe
```
allSubscriptions.reduce(({ name, claim_id: claimId }, index, arr) => {
  name && claimId && arr.push(`lbry://${name}#${claimId}`)
}, []);
```

Up to you